File: client/src/app/shared/models/order.ts
export interface Order {
    id: number
    orderDate: string
    updatedAt: string
    buyerEmail: string
    orderNumber?: string
    shippingAddress: ShippingAddress
    deliveryMethod: string
    shippingPrice: number
    paymentSummary?: PaymentSummary
    orderItems: OrderItem[]
    subtotal: number
    discount?: number
    currency: string
    status: string
    paymentType: string
    paymentStatus: string
    deliveryStatus: string
    paymentIntentId?: string
    invoiceNumber?: string
    specialNotes?: string
    voucherCode?: string
    // couponCode?: string
    appliedDiscountType?: string
    tracking?: OrderTracking
    comments: OrderComment[]
    auditLogs: OrderAuditLog[]
    refundAmount?: number
    refundedAt?: string
    total: number
  }
  
  export interface ShippingAddress {
    name: string
    line1: string
    line2?: string
    city: string
    postalCode: string
    country: string
  }
  
  export interface PaymentSummary {
    last4: number
    brand: string
    expMonth: number
    expYear: number
  }
  
  export interface OrderItem {
    productId: number
    productName: string
    pictureUrl: string
    price: number
    quantity: number
  }

  export interface OrderTracking {
    courierName?: string
    trackingNumber?: string
    trackingUrl?: string
    estimatedDeliveryDate?: string
  }

  export interface OrderComment {
    id: number
    authorEmail: string
    content: string
    isInternal: boolean
    createdAt: string
  }

  export interface OrderAuditLog {
    id: number
    userEmail: string
    timestamp: string
    fieldChanged: string
    oldValue?: string
    newValue?: string
    comment?: string
    action: string
  }
  
  export interface OrderToCreate {
    cartId: string
    deliveryMethodId: number
    shippingAddress: ShippingAddress
    paymentSummary?: PaymentSummary
    discount?: number
    paymentType: string
    specialNotes?: string
    voucherCode?: string
    // couponCode?: string
  }

  export interface UpdateOrderStatusDto {
    orderStatus?: string
    paymentStatus?: string
    deliveryStatus?: string
    comment?: string
    sendEmailNotification?: boolean
  }

  export interface OrderTrackingDto {
    courierName?: string
    trackingNumber?: string
    trackingUrl?: string
    estimatedDeliveryDate?: string
  }

  export interface AddCommentDto {
    content: string
    isInternal: boolean
  }

  export interface SendEmailDto {
    emailType: string
    oldValue?: string
    adminNotes?: string
  }

File: Core/Entities/OrderAggregate/Order.cs
using System;
using Core.Interfaces;
using Core.Enums;

namespace Core.Entities.OrderAggregate;

public class Order : BaseEntity, IDtoConvertible
{
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public required string BuyerEmail { get; set; }
    public string? OrderNumber { get; set; }
    public ShippingAddress ShippingAddress { get; set; } = null!;
    public DeliveryMethod DeliveryMethod { get; set; } = null!;
    public PaymentSummary? PaymentSummary { get; set; }
    public List<OrderItem> OrderItems { get; set; } = [];
    public decimal Subtotal { get; set; }
    public decimal Discount { get; set; }
    public string Currency { get; set; } = "USD";
    
    public OrderStatus Status { get; set; } = OrderStatus.New;
    public PaymentType PaymentType { get; set; } = PaymentType.Stripe;
    public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Pending;
    public DeliveryStatus DeliveryStatus { get; set; } = DeliveryStatus.Pending;
    
    public string? PaymentIntentId { get; set; }
    public string? InvoiceNumber { get; set; }
    public string? SpecialNotes { get; set; }
    
    // Voucher/Coupon tracking
    public string? VoucherCode { get; set; }
    // public string? CouponCode { get; set; }
    public string? AppliedDiscountType { get; set; } // "Voucher" or "Coupon"
    
    public OrderTracking? Tracking { get; set; }
    public List<OrderComment> Comments { get; set; } = [];
    public List<OrderAuditLog> AuditLogs { get; set; } = [];
    
    public decimal? RefundAmount { get; set; }
    public DateTime? RefundedAt { get; set; }

    // Guest checkout fields
    public bool IsGuestOrder { get; set; }
    public string? GuestName { get; set; }
    public string? GuestEmail { get; set; }
    public string? GuestPhone { get; set; }

    public decimal GetTotal() 
    {
        return Subtotal - Discount + DeliveryMethod.Price;
    }
}
